src/main/java/com/fasterxml/jackson/databind/jdk14/JDK14Util.java

```diff
@@ -91,10 +104,10 @@ public String[] getRecordFieldNames(Class<?> recordType) throws IllegalArgumentE
             for (int i = 0; i < components.length; i++) {
                 try {
                     names[i] = (String) RECORD_COMPONENT_GET_NAME.invoke(components[i]);
-                } catch (Exception e) {
+                } catch (Throwable t) {
```

Comment:
What would be the reason to catch Errors too?

Reply:
it's a required change unfortunately because invoke throws Throwable. this method shouldnt throw at all though

Reply:
Ok. Odd, not sure why it does that... never understood why method signature would do that (given `Error`s need not be declared anyway so why not just `throws Exception`). Oh well. At least this is for a reason then.

Reply:
Did you know that Java reflection API is now implemented using MethodHandle(s)? So by using Java reflection, you are already using MethodHandle(s) under the hood.

Reply:
Is this what you mean?
https://openjdk.org/jeps/416

I'll close this PR anyway because of the animal sniffer issues.